cni/pkg/ebpf/server/redirectionServer.go

```diff
@@ -0,0 +1,542 @@
+				multiErr = concatError(multiErr, err)
+			}
+		}
+	}
+	return multiErr
+}
+
+// temp usage, will remove when integration with CNI control plane
+func (r *RedirectServer) Redirect(w http.ResponseWriter, req *http.Request) {
+	fmt.Println("Get new post request")
+	ifindex, _ := strconv.Atoi(req.URL.Query().Get("i"))
+	peerIndex := 2
+	namespace := req.URL.Query().Get("n")
+	ztunnel, _ := strconv.ParseBool(req.URL.Query().Get("z"))
+	remove, _ := strconv.ParseBool(req.URL.Query().Get("r"))
+	ip := req.URL.Query().Get("p")
+	mac := req.URL.Query().Get("m")
+	args := RedirectArgs{ifindex: ifindex, peerIndex: peerIndex, ztunnel: ztunnel, namespace: namespace, remove: remove, macAddr: mac, ipAddr: ip}
+	r.AcceptRequest(&args)
+	fmt.Fprintf(w, "Applied %+v\n", args)
+}
+
+type ztunnelInfo struct {
+	Ifindex uint32 `json:"ifindex"`
+	MacAddr string `json:"mac_addr"`
+}
+
+// temp usage, will remove when integration with CNI control plane
+//
+//nolint:errcheck
+func (r *RedirectServer) DumpZtunnelInfo(w http.ResponseWriter, req *http.Request) {
+	v, err := r.dumpZtunnelInfo()
+	if err != nil {
+		w.WriteHeader(http.StatusServiceUnavailable)
+		w.Write([]byte(err.Error()))
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	var info ztunnelInfo
+	info.Ifindex = v.Ifindex
+	info.MacAddr = net.HardwareAddr(v.MacAddr[:]).String()
+	bytes, _ := json.Marshal(info)
+	w.Write(bytes)
+}
+
+type appInfo struct {
+	IPAddr  string `json:"ipaddr"`
+	Ifindex uint32 `json:"ifindex"`
+	MacAddr string `json:"mac_addr"`
+}
+
+// temp usage, will remove when integration with CNI control plane
+//
+//nolint:errcheck
+func (r *RedirectServer) DumpAppInfo(w http.ResponseWriter, req *http.Request) {
+	k, v := r.dumpAppInfo()
+	if len(k) != len(v) {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	info := make([]appInfo, len(k))
+	for i := range k {
+		ip := htonl(k[i])
+		info[i].IPAddr = net.IPv4(byte(ip>>24), byte(ip>>16), byte(ip>>8), byte(ip)).String()
+		info[i].Ifindex = v[i].Ifindex
+		info[i].MacAddr = net.HardwareAddr(v[i].MacAddr[:]).String()
+	}
+	bytes, _ := json.Marshal(info)
+	w.Write(bytes)
+}
+
+func (r *RedirectServer) AcceptRequest(redirectArgs *RedirectArgs) {
+	r.redirectArgsChan <- redirectArgs
+}
+
+func (r *RedirectServer) attachTCForZtunnel(ifindex, peerIndex uint32, namespace string) error {
+	// attach to ztunnel host ingress
+	if err := r.attachTC("", ifindex, "ingress", uint32(r.ztunnelHostingressFd), r.ztunnelHostingressProgName); err != nil {
+		return err
+	}
+	// attach to ztunnel ingress
+	if err := r.attachTC(namespace, peerIndex, "ingress", uint32(r.ztunnelIngressFd), r.ztunnelIngressProgName); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *RedirectServer) detachTCForZtunnel(ifindex, peerIndex uint32, namespace string) error {
+	// detach from ztunnel host ingress
+	if err := r.delClsactQdisc("", ifindex); err != nil {
```

Comment:
very nit: namign about is attachTC but to detach its delClsactQdisc. Could be dettachTc?